opensearch_py_ml/ml_commons/rag_pipeline/rag/IAMRoleHelper.py

```diff
@@ -0,0 +1,220 @@
+            iam_client.put_role_policy(
+                RoleName=role_name,
+                PolicyName='InlinePolicy',  # you can replace this with your preferred policy name
+                PolicyDocument=json.dumps(inline_policy_json)
+            )
+
+            print(f'Created role: {role_name}')
+            return role_arn
+
+        except ClientError as e:
+            print(f"Error creating the role: {e}")
+            return None
+
+    def get_role_arn(self, role_name):
+        if not role_name:
+            return None
+        iam_client = boto3.client('iam')
+        try:
+            response = iam_client.get_role(RoleName=role_name)
+            # Return ARN of the role
+            return response['Role']['Arn']
+        except ClientError as e:
+            if e.response['Error']['Code'] == 'NoSuchEntity':
+                print(f"The requested role {role_name} does not exist")
+                return None
+            else:
+                print(f"An error occurred: {e}")
+                return None
+
+    def get_role_details(self, role_name):
+        iam = boto3.client('iam')
+
+        try:
+            response = iam.get_role(RoleName=role_name)
+            role = response['Role']
+
+            print(f"Role Name: {role['RoleName']}")
+            print(f"Role ID: {role['RoleId']}")
+            print(f"ARN: {role['Arn']}")
+            print(f"Creation Date: {role['CreateDate']}")
+            print("Assume Role Policy Document:")
+            print(json.dumps(role['AssumeRolePolicyDocument'], indent=4, sort_keys=True))
+
+            list_role_policies_response = iam.list_role_policies(RoleName=role_name)
+
+            for policy_name in list_role_policies_response['PolicyNames']:
+                get_role_policy_response = iam.get_role_policy(RoleName=role_name, PolicyName=policy_name)
+                print(f"Role Policy Name: {get_role_policy_response['PolicyName']}")
+                print("Role Policy Document:")
+                print(json.dumps(get_role_policy_response['PolicyDocument'], indent=4, sort_keys=True))
+
+        except ClientError as e:
+            if e.response['Error']['Code'] == 'NoSuchEntity':
+                print(f'Role {role_name} does not exist.')
+            else:
+                print(f"An error occurred: {e}")
+
+    def get_user_arn(self, username):
+        if not username:
+            return None
+        iam_client = boto3.client('iam')
+
+        try:
+            response = iam_client.get_user(UserName=username)
+            user_arn = response['User']['Arn']
+            return user_arn
+        except ClientError as e:
+            if e.response['Error']['Code'] == 'NoSuchEntity':
+                print(f"IAM user '{username}' not found.")
+                return None
+            else:
+                print(f"An error occurred: {e}")
+                return None
+
+    def assume_role(self, role_arn, role_session_name="your_session_name"):
```

Comment:
The role_session_name is needed to uniquely identify the session when assuming a role. Using a default like 'your_session_name' is just a placeholder. It might be better to make this a required input or generate a unique name automatically to avoid any confusion.